<a href="https://colab.research.google.com/github/zaynsidd/QBIO_401/blob/main/QBIO_401_Assignment1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Question 1
def base_count(seq):
  tracker = [0]*4
  for char in seq:
    if char == 'A':
      tracker[0] += 1
    elif char == 'C':
      tracker[1] += 1
    elif char == 'G':
      tracker[2] += 1
    elif char == 'T':
      tracker[3] += 1
  return tracker

In [ ]:
#Question 2
def pair_count(seq, pair):
  tracker = 0
  for i in range(len(seq)-1):
    if seq[i:i+2] == pair:
      tracker += 1
  return tracker

In [ ]:
#Question 3
def double_base_count(seq):
  tracker = [0]*16
  bases = ['AA', 'AC', 'AG', 'AT', 'CA', 'CC', 'CG', 'CT',
           'GA', 'GC', 'GG', 'GT', 'TA', 'TC', 'TG', 'TT']
  bases_counter = dict(zip(bases, tracker))
  for i in range(len(seq)-1):
    bases_counter[seq[i:i+2]] += 1
  bclist = list(bases_counter.values())
  return bclist

In [ ]:
def loadFASTA(filename):
    '''Outputs a sequence string from the FASTA file named filename'''
    infile = open(filename) # opens the file
    seqlist = [] # empty list
    temp = infile.readline() # reads a single line
                             # we don't want the first line
                             # of the FASTA file ">..."
    for line in infile: # reads one line at a time
                        # when it gets to the end of the file it stops
        temp = line.replace("\n","") # removes \n
        temp = temp.replace("\r","") # removes \r too
        seqlist.append(temp)
    infile.close() # closes the file
    seq = "".join(seqlist) # combines the list into a string
    return seq

In [ ]:
seq1 = loadFASTA('/content/qbio_401_drosophila2_hw1.fasta')
seq2 = loadFASTA('/content/qbio_401_sequence_hw1.fasta')

In [ ]:
base_count(seq1)

[2395, 1876, 1675, 1718]

In [ ]:
base_count(seq2)

[1773, 1139, 1410, 1751]

In [ ]:
double_base_count(seq1)

[870,
 521,
 481,
 523,
 621,
 452,
 455,
 348,
 499,
 501,
 323,
 352,
 404,
 402,
 416,
 495]

In [ ]:
double_base_count(seq2)

[595, 280, 459, 438, 394, 275, 99, 371, 443, 276, 355, 336, 340, 308, 497, 606]

Question 4:
Yes, we see this. Sequence 1 (Drosophila) has 455 whereas sequence 2 has only 99.
This is indicated by the 7th amount in each list.